homeassistant/components/vicare/fan.py

```diff
@@ -0,0 +1,172 @@
+"""Viessmann ViCare ventilation device."""
+
+from __future__ import annotations
+
+from contextlib import suppress
+import logging
+from typing import Any
+
+from PyViCare.PyViCareDevice import Device as PyViCareDevice
+from PyViCare.PyViCareDeviceConfig import PyViCareDeviceConfig
+from PyViCare.PyViCareUtils import (
+    PyViCareInvalidDataError,
+    PyViCareNotSupportedFeatureError,
+    PyViCareRateLimitError,
+)
+from PyViCare.PyViCareVentilationDevice import (
+    VentilationDevice as PyViCareVentilationDevice,
+)
+from requests.exceptions import ConnectionError as RequestConnectionError
+
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    ordered_list_item_to_percentage,
+    percentage_to_ordered_list_item,
+)
+
+from .const import DEVICE_LIST, DOMAIN
+from .entity import ViCareEntity
+from .types import VentilationMode, VentilationProgram, ViCareDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+ORDERED_NAMED_FAN_SPEEDS = [
+    VentilationProgram.LEVEL_ONE,
+    VentilationProgram.LEVEL_TWO,
+    VentilationProgram.LEVEL_THREE,
+    VentilationProgram.LEVEL_FOUR,
+]
+
+PRESET_MODES = [
+    VentilationMode.PERMANENT,
+    VentilationMode.VENTILATION,
+    VentilationMode.SENSOR_DRIVEN,
+    VentilationMode.SENSOR_OVERRIDE,
+]
+
+
+def _build_entities(
+    device_list: list[ViCareDevice],
+) -> list[ViCareFan]:
+    """Create ViCare fan entities for a device."""
+
+    return [
+        ViCareFan(device.config, device.api, "ventilation")
+        for device in device_list
+        if isinstance(device.api, PyViCareVentilationDevice)
+    ]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the ViCare fan platform."""
+
+    device_list = hass.data[DOMAIN][config_entry.entry_id][DEVICE_LIST]
+
+    async_add_entities(
+        await hass.async_add_executor_job(
```

Comment:
Please move the side effects out of the constructor for the entity and into into the function that runs in the executor.

We want to avoid side effects in `__init__`

Reply:
commit dddf4b96462b9a7b6836dc06f5d9ed5918398881 removed the side effects from the constructor, I think you no longer need to make the objects in an executor thread?